docs/reference/extensions/flask-framework-extension.md

```diff
@@ -204,6 +218,46 @@ in the application, the services with the name ending in `-worker` will run in a
 only run in one of the units of the application.
 
 
+## Observability
+
+12-Factor charms are designed to be easily observable using the [Canonical Observability Stack](https://charmhub.io/topics/canonical-observability-stack).
+
+You can easily integrate your charm with [Loki](https://charmhub.io/loki-k8s) and [Prometheus](https://charmhub.io/prometheus-k8s) using Juju.
+
+```shell
+juju relate flask-k8s grafana
+juju relate flask-k8s loki
+juju relate flask-k8s prometheus
+```
+After integration, you will be able to observe your workload using Grafana dashboards.
+
+
+In addition to that you can also trace your workload code using [Tempo](https://charmhub.io/topics/charmed-tempo-ha).
+To learn about how to deploy Tempo you can read the documentation [here](https://charmhub.io/topics/charmed-tempo-ha).
+
+To enable tracing in your Flask app you can simply add the following lines to your workload code:
+
+```python
+from opentelemetry import trace
+from opentelemetry.instrumentation.flask import FlaskInstrumentor
+
+app = Flask(__name__)
+FlaskInstrumentor().instrument_app(app)
+tracer = trace.get_tracer(__name__)
+```
+
+You also need to add the following lines into your `requirements.txt` file:
+
+```
+opentelemetry-api
+opentelemetry-exporter-otlp-proto-http
+opentelemetry-instrumentation-flask
+opentelemetry-sdk
+```
```

Comment:
Let's do this in the how to

Reply:
Actually, instead of a how to, please prepare a blog post I can attach to the regular updates about tracing in Flask. We can include links to the frameworks here for people to read more

Reply:
We should do a spread test for flask